force-app/main/default/lwc/newServiceSchedule/newServiceSchedule.js

```diff
@@ -0,0 +1,165 @@
+        seriesEnds: SERVICE_SECHEDULE_ENDS_FIELD.fieldApiName,
+    };
+
+    @wire(getFieldSet, {
+        objectName: SERVICE_SCHEDULE_OBJECT.objectApiName,
+        fieldSetName: FIELD_SET_NAME,
+    })
+    wiredFieldSet({ error, data }) {
+        if (data) {
+            this.setFieldSet(data);
+        } else if (error) {
+            console.log(JSON.stringify(error));
+        }
+    }
+
+    setFieldSet(data) {
+        const recurrenceFields = Object.values(this.dateFields);
+        recurrenceFields.push(Object.values(this.picklistFields));
+
+        this.fieldSet = data
+            .filter(obj => !recurrenceFields.includes(obj.apiName))
+            .map(obj => {
+                let field = { ...obj };
+                field.size =
+                    obj.apiName === NAME_FIELD.fieldApiName ? LARGE_SIZE : SMALL_SIZE;
+                return field;
+            });
+    }
+
+    @wire(getObjectInfo, { objectApiName: SERVICE_SCHEDULE_OBJECT })
+    wiredObject(result) {
+        if (!result) {
+            return;
+        }
+
+        if (result.data) {
+            this.recordTypeId = this.recordTypeId
+                ? this.recordTypeId
+                : result.data.defaultRecordTypeId;
+            this.fields = result.data.fields;
+            this.isLoaded = this.wiredPicklistValues !== undefined;
+        } else if (result.error) {
+            console.log(JSON.stringify(result.error));
+        }
+    }
+
+    @wire(getPicklistValuesByRecordType, {
+        objectApiName: SERVICE_SCHEDULE_OBJECT,
+        recordTypeId: "$recordTypeId",
+    })
+    wiredPicklists(result) {
+        if (!result) {
+            return;
+        }
+        this.picklists = {};
+
+        if (result.data) {
+            Object.keys(this.picklistFields).forEach(field => {
+                const fieldApiName = this.picklistFields[field];
+                const picklistField = {
+                    ...result.data.picklistFieldValues[fieldApiName],
+                };
+
+                if (picklistField && this.fields[fieldApiName]) {
+                    picklistField.label = this.fields[fieldApiName].label;
+                    this.picklists[field] = picklistField;
+                }
+            });
+            this.isLoaded = this.fields !== undefined;
+        }
+    }
+
+    handleFrequencyChange(event) {
+        if (!event || !event.detail) {
+            return;
+        }
+
+        this.selectedFrequency = event.detail.length ? event.detail[0].value : undefined;
+        this.isWeekly = this.selectedFrequency === WEEKLY;
+        this.isRecurring = this.selectedFrequency !== ONE_TIME;
+    }
+
+    handleDaysOfWeekChange(event) {
+        if (!event || !event.detail) {
+            return;
+        }
+
+        this.selectedDaysOfWeek = event.detail.length
+            ? event.detail.map(selection => selection.value)
+            : undefined;
+    }
+
+    handleSeriesEndsChange(event) {
+        if (!event || !event.detail) {
```

Comment:
when would this ever come back null?

Reply:
After the call, we decided to not check this and feel confident it would not be null.